doc/config.rst

```diff
@@ -141,3 +141,20 @@ configure them in your config file.
         ->setIndent("\t")
         ->setLineEnding("\r\n")
     ;
+
+It's possible to register custom rule sets, which makes it easier to reuse custom configuration between multiple projects. If you have prepared rule set, you can register it and then enable it in the rules. Custom rule sets (in this example ``\MyNameSpace\MyRuleSetClass``) must implement ``\PhpCsFixer\RuleSet\RuleSetDescriptionInterface``.
+
+.. code-block:: php
+
+    <?php
+
+    return (new PhpCsFixer\Config())
+        ->registerCustomRuleSets([
+            MyNameSpace\MyRuleSetClass::class, // It identifies itself as '@MyRuleSet'
+        ])
+        ->setRules([
+            '@MyRuleSet' => true,
+        ])
+    ;
+
+ℹ️ If you use other ``\PhpCsFixer\ConfigInterface`` implementation than built-in one, you can register custom rule sets using ``\PhpCsFixer\RuleSet\RuleSets::registerRuleSet()`` just before the configuration. Currently ``\PhpCsFixer\Config::registerCustomRuleSets()`` is not part of the ``ConfigInterface`` because it would be a BC-break - it will be added there in the next major version of the Fixer.
```

Comment:
Alternative approach is to introduce additional interface and implement it in default `Config` class, just like it's done in #7777 with `ParallelRunnerConfigInterface` 🙂. This way we would have out-of-the-box support for it in our config implementation, and anyone using custom config classes could implement this interface too. Then `RuleSets::registerRuleSet()` wouldn't be needed anymore.

Reply:
It was done with `CustomRulesetsAwareConfigInterface`.